crates/ruff_python_ast/src/nodes.rs

```diff
@@ -767,12 +767,37 @@ impl From<ExprIf> for Expr {
     }
 }
 
+#[derive(Debug, Clone, PartialEq)]
+pub struct DictItem {
+    pub key: Option<Expr>,
+    pub value: Expr,
+}
```

Comment:
I wonder if you thought about an alternative design which would utilize an enum like:
```rs
pub enum DictItem {
	KeyValue(KeyValueItem),
	DoubleStarred(Expr), // or Expanded, Spread, ...
}
```

Although this approach does add in an additional indirection.

That said, I'd keep the `DictItem` approach for now.

Reply:
Oh, very nice... I hadn't considered this at all! And I think it makes it much more readable.

Reply:
Just to be clear, I'd keep the `DictItem` approach as is in this PR. This is just an alternative design that I had in mind while working on the parser :)